interfaces/builtin/zigbee_dongle.go

```diff
@@ -0,0 +1,137 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2016 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+import (
+	"bytes"
+	"fmt"
+	"path/filepath"
+
+	"github.com/snapcore/snapd/interfaces"
+)
+
+// ZigbeeDongleInterface is the type for serial port interfaces.
+type ZigbeeDongleInterface struct{}
+
+// Name of the zigbee-dongle interface.
+func (iface *ZigbeeDongleInterface) Name() string {
+	return "zigbee-dongle"
+}
+
+func (iface *ZigbeeDongleInterface) String() string {
+	return iface.Name()
+}
+
+var zigbeeDongleDevPath = "/dev/zigbee/*"
+
+var zigbeeDonglePermanentSlotUdev = []byte(`
```

Comment:
This cannot be used this way. 

As mentioned in private. Currently udev rules are *per app* so this would cause this hook to run multiple times. I would suggest creating a new backend or expanding this backend to say "this slot is not associated with applications" so that the emitted snippet is showing up exactly once.